PictureApp.API/Controllers/UsersController .cs

```diff
@@ -15,25 +15,73 @@ public class UsersController : ControllerBase
     {
         private readonly IUserService _userService;
         private readonly IMapper _mapper;
+        private readonly IPhotoService _photoService;
 
-        public UsersController(IUserService userService, IMapper mapper)
+        public UsersController(IUserService userService, IMapper mapper, IPhotoService photoService)
         {
             _userService = userService ?? throw new ArgumentNullException(nameof(userService));
             _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            _photoService = photoService;
         }
 
         [HttpGet("{id}")]
-        public IActionResult GetUser(int id)
+        public async Task<IActionResult> GetUser(int id)
         {
-            return Ok(_userService.GetUser(id));
+            try
+            {
+                CheckIfCurrentUserIsCorrect(id);
+
+                return Ok(await _userService.GetUser(id));
+            }
+            catch(Exception)
+            {
+                return BadRequest();
+            }
         }
 
-        [HttpGet("allUserWithFollowerInfo")]
-        public async Task<IActionResult> GetUsersWithFollowers()
+        [HttpGet("userEditProfile/{id}")]
+        public async Task<IActionResult> GetUserForEdit(int id)
```

Comment:
This will work only for current user. Is this our purpose? If so, You can obtain user from User static class instead.